VONK-2590: Update release note for v4.2.0 and v4.2.1 (#9)
FirelyTeam/firely-docs-firely-server: releasenotes/releasenotes.rst

The default FHIR version of the `public Firely Server endpoint <https://server.fire.ly/>`_ is now R4.

> .. _vonk_releasenotes_421:

Reviewer: I would add a note saying that we needed to fix due to an error in 4.2.0

Reviewer: Also add a comment about what was wrong in 4.2.0

Author: Thanks for comments. I have updated the release notes. Could you check it again? Please feel free to suggest otherwise. :-)

Author: I'll mark this as resolved since Lennart is helping already.

The default FHIR version of the `public Firely Server endpoint <https://server.fire.ly/>`_ is now R4.

.. _vonk_releasenotes_421:

Release 4.2.1 hotfix
--------------------

Database
^^^^^^^^
.. note::
   We found an issue in version 4.2.0, which affects the query performance for Firely Server running on a SQL Server database. If your are running FS v4.2.0 on SQL Server you should upgrade to v4.2.1 or if that is not possible, :ref:`vonk-contact`.

.. attention::
    The upgrade procedure will execute a SQL script try to validate the foreign key constraints. If your database is large, this may take too long and the upgrade process will time out. If that happens you need to run the upgrade script manually, The script can be found in ``data/20210720085032_EnableCheckConstraintForForeignKey.sql``. Here are some guidelines:

   * We tested it on a database with about 15k Patient records, and 14 million resources in total. The upgrade script took about 20 seconds to complete on a fairly powerful laptop.

>    * As always, make sure you have a backup of your database that has tried and tested before you begin the upgrade.

Reviewer 2: ```suggestion
   * As always, make sure you have a backup of your database that has been tried and tested before you begin the upgrade.
```

   * If you already expect the migration might failed or time out, you can run it manually upfront. Shut down Firely Server, so no other users are using the database, and then run the script from SQL Server Management Studio (or a similar tool).

Fix
^^^

> #. Fixed a bug about Not Trusted Foreign Key Constraints. This bug has an impact on the query performance since SQL Server optimizer won't trust these constants and to perform extra checks.

Reviewer 2: ```suggestion
#. Fixed a bug where some of the Foreign Keys in SQL Server had become untrusted. This bug has an impact on the query performance since the the SQL Server query optimizer will not consider FKs when they are not trusted. This has been fixed, all Foreign Keys have been validated and are trusted again.
```

> .. attention::

Reviewer 2: ```suggestion
.. attention::
   For SQL Server users: this version of Firely Server running on SQL Server has a bug where some of the Foreign Keys became untrusted. This has an impact on the query performance. Please upgrade to version 4.2.1 or if that is not possible, :ref:`vonk-contact`.
   Please note that users running Firely Server running either MongoDb, CosmoDb, or SQLite are not affected by this issue.

.. attention::
```

   * We tested it on a database with about 15k Patient records, and 14 mln resources in total. Migrating that took about 50 minutes on a fairly powerful laptop.
   * Absolutely make sure you create a backup of your database first.
   * If you haven't done so already, first upgrade to version 4.1.x.
   * If you already expect the migration might time out, you can run it manually upfront. Shut down Firely Server, so no other users are using the database, and then run the script from SQL Server Management Studio (or a similar tool).
   * Running the second script (`20210520102224_ChangePrimaryKeyTypeFromIntToBigintBDE.sql`) is optional - that should also succeed when applied by the automigration.

.. note::
   As reported by a client, for this version of Firely Server, some foreign keys are untrusted in the SQL Server database. This has an impact on query performance and referential integrity. If you find yourself in the same situation, please use the script provided in version 4.2.1 to fix the issue.

> 

Reviewer 2: I have moved this section all the way to the top of 4.2.0 release notes.

Author: Thanks a lot Lennart, the wordings are much better. I've remove this "note" section.
